### PR TITLE
Location Change Event

### PR DESCRIPTION
**Pull request by Ghost-3**
Event is invoked when parsing `/locraw` if the new location is different from the old one.

### CODE REVIEW

File: src/main/java/de/hysky/skyblocker/events/LocationEvents.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    public static final Event<LocationChange> CHANGE =
+            EventFactory.createArrayBacked(LocationChange.class, callbacks -> location -> {
+                for (LocationChange callback : callbacks)
+                    callback.onLocationChange(location);
+            });
+
+    @Environment(EnvType.CLIENT)
+    @FunctionalInterface
+    public interface LocationChange {
+        void onLocationChange(Location location);
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/events/LocationEvents.java:20]: Rename the `location` parameter to better describe what information it provides as it's a big ambiguous as to whether its the new location or the old one.

**Review by AzureAaron — CHANGES_REQUESTED** [review]

**Comment by Ghost-3** [comment]
Fixed

**Review by AzureAaron — APPROVED** [review]

File: src/main/java/de/hysky/skyblocker/utils/Utils.java [hunk 2]
```diff
@@ -384,13 +385,16 @@ private static void updateLocRaw() {
 
     /**
      * Parses /locraw chat message and updates {@link #server}, {@link #gameType}, {@link #locationRaw}, {@link #map}
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/utils/Utils.java:387]:
```suggestion
     * Parses /locraw chat message, updates {@link #server}, {@link #gameType}, {@link #locationRaw}, {@link #map}, and {@link #location},
```

File: src/main/java/de/hysky/skyblocker/utils/Utils.java [hunk 3]
```diff
@@ -384,13 +385,16 @@ private static void updateLocRaw() {
...
-     * and {@link #location}
+     * and {@link #location}.
+     * It also invokes location change event.
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/utils/Utils.java:389]:
```suggestion
     * and invokes {@link LocationEvents#CHANGE}.
```

File: src/main/java/de/hysky/skyblocker/events/LocationEvents.java
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/events/LocationEvents.java:1]: Maybe just put this event in `SkyblockEvents` so less class spam.

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]

**Comment by viciscat** [comment]
Superseded by #716

**Pull request closed without merging**